.github/workflows/smoke-test.yml

```diff
@@ -199,7 +199,9 @@ jobs:
         # We only expect test coverage with all extra dependencies.
         # "polars" implies "scikit-learn": see setup.cfg.
         if: ${{ contains(matrix.extra-install, 'polars') && contains(matrix.python-version, '3.12') }}
-        run: coverage report | tee /tmp/coverage.txt
+        run: |
+          set -o pipefail
+          coverage report | tee /tmp/coverage.txt
```

Comment:
I remember adding the `tee`: Sorry that it broke things!

I had considered just redirecting the output to a file, but then you don't see the output if it fails, so this seems like the best solution.